syntax = "proto3";
package lastmile.v1;
option go_package = "github.com/yourorg/lastmile/api/gen/go/lastmile/v1;lastmilev1";
import "lastmile/v1/common.proto";

service UserService {
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
  rpc Authenticate(AuthenticateRequest) returns (AuthenticateResponse);
}

message CreateUserRequest { User user = 1; string password = 2; }
message CreateUserResponse { User user = 1; }
message GetUserRequest { string id = 1; }
message GetUserResponse { User user = 1; }
message AuthenticateRequest { string phone = 1; string password = 2; }
message AuthenticateResponse { string user_id = 1; string jwt = 2; }
